packages/react-components/src/components/menu/positionable.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import { usePopper } from 'react-popper';
+import preventOverflow from '@popperjs/core/lib/modifiers/preventOverflow.js';
+import flip from '@popperjs/core/lib/modifiers/flip.js';
+import type { Options } from '@popperjs/core/lib/types';
+import type { PropsWithChildren, RefCallback } from 'react';
+
+import { Menu, MenuProps } from './menu';
+
+import './menu.css';
+
+type Position = { x: number; y: number; z?: number };
+type ReferenceElement = (setRef: RefCallback<null>) => React.ReactNode;
+
+type AbsoluteMenu = {
+  position: Position;
+  referenceElement?: never;
+};
+type RelativeMenu = {
+  position?: never;
```

Comment:
curious .. why never ?

Reply:
using this to indicate an XOR type for positioning. It can either be relative or absolute. relative uses a reference element, absolute uses a position